diffusion/train.py

```diff
@@ -65,14 +65,17 @@ def train(config: DictConfig) -> None:
     # Check if this is training an autoencoder. If so, the optimizer needs different param groups
     if hasattr(model, 'autoencoder_loss'):
         optimizer = make_autoencoder_optimizer(config, model)
+        tokenizer = None
     else:
         optimizer = hydra.utils.instantiate(config.optimizer, params=model.parameters())
+        tokenizer = model.tokenizer
 
     # Load train dataset. Currently this expects to load according to the datasetHparam method.
     # This means adding external datasets is currently not super easy. Will refactor or check for
     # upstream composer changes that could make this easier.
     train_dataloader: Union[Iterable, DataSpec, Dict[str, Any]] = hydra.utils.instantiate(
         config.dataset.train_dataset,
+        tokenizer=tokenizer,
```

Comment:
I think it might be a good idea to only pass tokenizer to the dataloader if `model.tokenizer` exists. Reason being we also have an image only dataset [here](https://github.com/mosaicml/diffusion/blob/da716d7c424d58f1d8e6f2e91c6f0b4029ea53ae/diffusion/datasets/image.py#L21) that doesn't take a tokenizer, which is what one _should_ use for autoencoder training but the yamls are out of data (my fault, sorry...)

Any other solution that doesn't require dataloaders to accept a tokenizer argument would also work.

Reply:
Added 3 ugly If/else branches =(. 

I tried to get around this by adding the tokenizer directly to the omegaconf config, but omegaconf doesn't like having random giant objects added to it.

the functionality should be there now though. 

Also updated eval.py